spec.bs

```diff
@@ -850,6 +876,66 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
 
 </div>
 
+<div algorithm>
+To <dfn>build an interest group passed to generateBid</dfn> given an [=interest group=] |ig|:
+  
+  1. Let |igGenerateBid| be a new {{GenerateBidInterestGroup}} with the following fields:
+    <dl>
+      <dt>{{GenerateBidInterestGroup/owner}}
+      <dd>The [=serialization of an origin|serialization=] of |ig|'s [=interest group/owner=]
+      <dt>{{GenerateBidInterestGroup/name}}
+      <dd>|ig|'s [=interest group/name=]
+      <dt>{{GenerateBidInterestGroup/enableBiddingSignalsPrioritization}}
+      <dd>|ig|'s [=interest group/enable bidding signals prioritization=]
+      <dt>{{GenerateBidInterestGroup/priorityVector}}
+      <dd>|ig|'s [=interest group/priority vector=] if not null, otherwise {{undefined}}
+      <dt>{{GenerateBidInterestGroup/executionMode}}
+      <dd>|ig|'s [=interest group/execution mode=]
+      <dt>{{GenerateBidInterestGroup/biddingLogicURL}}
+      <dd>The [=serialize a URL|serialization=] of |ig|'s [=interest group/bidding url=]
+      <dt>{{GenerateBidInterestGroup/biddingWasmHelperURL}}
+      <dd>The [=serialize a URL|serialization=] of |ig|'s [=interest group/bidding wasm helper url=]
+      <dt>{{GenerateBidInterestGroup/updateURL}}
+      <dd>The [=serialize a URL|serialization=] of |ig|'s [=interest group/update url=]
+      <dt>{{GenerateBidInterestGroup/trustedBiddingSignalsURL}}
+      <dd>The [=serialize a URL|serialization=] of |ig|'s [=interest group/trusted bidding signals url=]
+      <dt>{{GenerateBidInterestGroup/trustedBiddingSignalsKeys}}
+      <dd>|ig|'s [=interest group/trusted bidding signals keys=]
+      <dt>{{GenerateBidInterestGroup/userBiddingSignals}}
+      <dd>[=Parsed a JSON string to a JavaScript value=] given |ig|'s [=interest group/user bidding signals=]
+      <dt>{{GenerateBidInterestGroup/ads}}
+      <dd>|ig|'s [=interest group/ads=] [=converted to an AuctionAd sequence=]
+      <dt>{{GenerateBidInterestGroup/adComponents}}
+      <dd>|ig|'s [=interest group/ad components=] [=converted to an AuctionAd sequence=]
+    </dl>
+  1. Return |igGenerateBid|.
+</div>
+
+<div algorithm>
+To <dfn>serialize a URL</dfn> given a [=URL=]-or-null |url|:
+
+  1. If |url| is null, then return {{undefined}}.
+  1. Return the [=URL serializer|serialization=] of |url|.
+</div>
+
+<div algorithm>
+To <dfn>convert to an AuctionAd sequence</dfn> given a [=list=]-or-null |ads|:
+
+  1. If |ads| is null, then return {{undefined}}.
+  1. Let |adsIDL| be a new [=list=].

```

Comment:
I would use
```suggestion
  1. Let |adsIDL| be a new <code>{{sequence}}&lt;{{AuctionAd}}></code>.
```
It means the same thing since https://webidl.spec.whatwg.org/#idl-sequence says sequences are lists, but it's clearer about what to expect.